### PR TITLE
MM-53989 Upgrade to Node 18 and NPM 9

### PR DESCRIPTION
**Pull request by hmhealey**
#### Summary
Node 16 has been out of support for quite a while now, and Node 18 is the current LTS (until October). Having the web app not be on the current LTS has caused a bunch of problems for community members and for people working on the mobile repo which requires Node 18 already.

I also took the opportunity to remove `skip_integrity_check.js` because NPM 9 automatically skips integrity checks for git dependencies! 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-53989

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-16.10.0
+18.17.1
```
Comment (crspeller) [inline .nvmrc:1]: Should we do `18.17` so we are on latest patch? Or is that too unstable for the JS ecosystem.

**Review by crspeller — APPROVED** [review]
Yay node and NPM upgrades! Can't wait to see how this breaks horribly 😆 Not going to block on my suggestion.

File: .nvmrc (hunk 1 above)
Comment (hmhealey) [reply .nvmrc:1]: Yep, that's fine by me. 18.17.1 is the current latest patch anyway.

We could even use just `18`, but I think it was some dot release of 18 that made NPM 9 the default, and I don't feel like looking up which one that was

```suggestion
18.17
```

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by saturninoabril — COMMENTED** [review]
Thanks @hmhealey! Did a quick check and qualification, and looks good to me. Holding until tomorrow (Friday) for readiness check on CI and Infra to lessen surprises - https://community.mattermost.com/private-core/pl/jpbf3m6frfnixpxnc5q1eb5eke.

- On a quick glance, most of the CI are using `.nvmrc` except the Cypress benchmark tests. Should it be changed here as well?

To follow separately, align node version of Cypress tests (full and smoke) which are currently at node@16. cc @mvitale1989 @yasserfaraazkhan

**Comment by hmhealey** [comment]
Let's hold it until Monday then just to avoid breaking everything before the weekend.

I forgot the Cypress benchmark tests weren't using that, but they've been commented out since the monorepo migration, so I wouldn't worry about those. We should probably just remove them since we never got them stable to begin with

**Review by saturninoabril — APPROVED** [review]
Thanks @hmhealey! Looks good to me.

Related MR submitted for build - https://git.internal.mattermost.com/mattermost/ci/build-ee/-/merge_requests/71

**Comment by hmhealey** [comment]
@mvitale1989 I'll add that in before I merge this, but is there a way to have that automatically use a version based on the `package.json` or `.nvmrc`? We're already stuck with 2 sources of truth for the Node version, and I'd rather not add a third since we used to have a ton of places that had the Node version and so it was almost always out of sync somewhere

**Pull request merged**